tools/git_utils.py

```diff
@@ -62,17 +62,38 @@ def get_latest_run(workflow: Workflow) -> WorkflowRun:
     raise RuntimeError("Unable to list all workflow invocations")
 
 
-def get_name_and_commit_hash(branch_name: Optional[str]) -> tuple[str, str]:
-    branch_name = branch_name or "HEAD"
-    name = (
-        subprocess.check_output(["git", "rev-parse", "--abbrev-ref", branch_name], stderr=subprocess.STDOUT)
+def get_name_and_commit_hash(local_branch_name: Optional[str]) -> tuple[str, str]:
+    local_branch_name = local_branch_name or "HEAD"
+    has_upstream = False
+
+    try:
+        # Check if the branch has a remote tracking branch.
+        upstream_branch = (
+            subprocess.check_output(
+                ["git", "rev-parse", "--abbrev-ref", f"{local_branch_name}@{{upstream}}"], stderr=subprocess.STDOUT
+            )
+            .strip()
+            .decode("utf-8")
+        )
+        has_upstream = True
+        local_branch_name = upstream_branch
```

Comment:
Instead of setting `has_upstream = True` here and then performing a check later, can't we just perform the splitting here? I.e.:

```py
local_branch_name = upstream_branch.split("/", 1)[1]
```

Reply:
Good catches, I was kind of lazy when coding this up 😅 cleaned up now!